tom_tns/forms.py

```diff
@@ -96,14 +135,123 @@ def __init__(self, *args, **kwargs):
             ),
             Row(Column('photometry_remarks')),
             Row(HTML('<h4>Last Nondetection</h4>')),
-            Row(
-                Column('archive'),
-                Column('archival_remarks'),
-                Column('nondetection_remarks', css_class='col-md-6'),
+            Alert(
+                content="""Fill out either the Archive information,
+                           or the Detection information for the last nondetection
+                        """,
+                css_class='alert-warning'
+            ),
+            Accordion(
+                AccordionGroup('Archive information',
+                               Row(
+                                    Column('archive'),
+                                    Column('archival_remarks'),
+                                  ),
+                               ),
+                AccordionGroup('Detection information',
+                               Row(
+                                    Column('nondetection_observation_date'),
+                                    Column('nondetection_instrument'),
+                                    Column('nondetection_exposure_time'),
+                                    Column('nondetection_observer'),
+                                  ),
+                               Row(
+                                    Column('nondetection_flux'),
+                                    Column('nondetection_flux_units'),
+                                    Column('nondetection_filter'),
+                                  ),
+                               Row(Column('nondetection_remarks', css_class='col-md-12'))
+                               )
             ),
             Row(Column(Submit('submit', 'Submit Report'))),
         )
 
+    def clean(self):
+        super().clean()
+        # Either the archival info or nondetection flux info must be set for a valid TNS submission
+        if not self.is_set('archive') or not self.is_set('archival_remarks'):
+            if any([not self.is_set(field) for field in [
+                'nondetection_flux', 'nondetection_instrument', 'nondetection_filter', 'nondetection_observation_date'
+            ]]):
+                raise ValidationError(
+                    "Must set either last nondetection archival information,"
+                    "or last nondetection flux, obsdate, filter and instrument"
+                )
+
+    def generate_hermes_report(self):
```

Comment:
I'm a little confused. Is generate hermes report defined twice? It shows up here and below.

Reply:
There are two Form classes, one for the new discovery form and one for the classification form. They both implement the same interface so they can be called polymorphically, but those functions each do something different depending on which form instance it is.